horace_core/sqw/@sqw/mask_random_pixels.m

```diff
@@ -25,20 +25,20 @@
 % Original author: S. Toth
 % Modifications: R. A. Ewings
 
-if ~has_pixels(win)
+if ~has_pixels(obj)
     error('HORACE:sqw:invalid_argument', ...
         'Can not mask random pixels on sqw object not containing any pixels')
 end
 
-wout = copy(win);
-sz = numel(win);
+wout = obj;
+sz = numel(obj);
 
 npix = round(npix);
-if isscalar(npix)
+if isscalar(npix) && sz>1
```

Comment:
because sz is the size of array of objects and npix -- the number of pixels to keep. 
we may want to keep the same number of pixels for each object.